File: official/docs/ruby/v5/orders/one-call-buy.rb
require 'easypost'

client = EasyPost::Client.new(api_key: ENV['EASYPOST_API_KEY'])

order = client.order.create(
  to_address: {
    id: 'adr_...',
  },
  from_address: {
    id: 'adr_...',
  },
  shipments: [
    {
      parcel: {
        weight: 10.2,
      },
    },
    {
      parcel: {
        predefined_package: 'FedExBox',
        weight: 17.5,
      },
    },
  ],
  service: 'NextDayAir',
  carrier_accounts: ['ca_...'],
)

puts order

File: official/docs/java/v5/orders/one-call-buy.java
package order;

import com.easypost.EasyPost;
import com.easypost.exception.EasyPostException;
import com.easypost.model.Order;
import java.util.ArrayList;
import java.util.HashMap;

public class OneCallBuy {
    public static void main(String[] args) throws EasyPostException {
        EasyPost.apiKey = "EASYPOST_API_KEY";

        HashMap<String, Object> toAddress = new HashMap<String, Object>();
        toAddress.put("id", "adr_...");

        HashMap<String, Object> fromAddress = new HashMap<String, Object>();
        fromAddress.put("id", "adr_...");

        HashMap<String, Object> firstParcel = new HashMap<String, Object>();
        firstParcel.put("weight", 10.2);

        HashMap<String, Object> secondParcel = new HashMap<String, Object>();
        secondParcel.put("weight", 17.5);
        secondParcel.put("predefined_package", "FedExBox");

        ArrayList<HashMap<String, Object>> shipments = new ArrayList<HashMap<String, Object>>();
        shipments.add(firstParcel);
        shipments.add(secondParcel);

        HashMap<String, Object> orderHash = new HashMap<String, Object>();
        orderHash.put("carrier_accounts", "ca_...");
        orderHash.put("service", "NextDayAir");
        orderHash.put("to_address", toAddress);
        orderHash.put("from_address", fromAddress);
        orderHash.put("shipments", shipments);

        Order order = Order.create(orderHash);

        System.out.println(order);
    }
}
